pkgs/development/libraries/tracker/default.nix

```diff
@@ -90,21 +91,29 @@ stdenv.mkDerivation rec {
     [
       "--cross-file=${crossFile}"
     ]
-  );
+  ) ++ lib.optionals (!stdenv.isLinux) [
+    "-Dsystemd_user_services=false"
+  ];
 
   doCheck = true;
 
-  preCheck = ''
-    # (tracker-store:6194): Tracker-CRITICAL **: 09:34:07.722: Cannot initialize database: Could not open sqlite3 database:'/homeless-shelter/.cache/tracker/meta.db': unable to open database file
-    export HOME=$(mktemp -d)
-
-    # Our gobject-introspection patches make the shared library paths absolute
-    # in the GIR files. When running functional tests, the library is not yet installed,
-    # though, so we need to replace the absolute path with a local one during build.
-    # We are using a symlink that will be overridden during installation.
-    mkdir -p $out/lib
-    ln -s $PWD/src/libtracker-sparql/libtracker-sparql-3.0.so $out/lib/libtracker-sparql-3.0.so.0
-  '';
+  preCheck =
+    let
+      linuxDot0 = lib.optionalString stdenv.isLinux ".0";
+      darwinDot0 = lib.optionalString stdenv.isDarwin ".0";
+      extension = stdenv.hostPlatform.extensions.sharedLibrary;
+    in
+    ''
+      # (tracker-store:6194): Tracker-CRITICAL **: 09:34:07.722: Cannot initialize database: Could not open sqlite3 database:'/homeless-shelter/.cache/tracker/meta.db': unable to open database file
+      export HOME=$(mktemp -d)
+
+      # Our gobject-introspection patches make the shared library paths absolute
+      # in the GIR files. When running functional tests, the library is not yet installed,
```

Comment:
Isn't install happening before checkPhase?

Reply:
Nope, that is only for `buildPythonPackage`, which uses the `checkPhase` name for `installCheckPhase`:

https://github.com/NixOS/nixpkgs/blob/23e0738cf3fafb70443e415c689be3501b43658d/pkgs/stdenv/generic/setup.sh#L1361-L1362
https://github.com/NixOS/nixpkgs/blob/23e0738cf3fafb70443e415c689be3501b43658d/pkgs/development/interpreters/python/mk-python-derivation.nix#L188